service/collector/conditions.go

```diff
@@ -0,0 +1,107 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/label"
+	aeconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ConditionsConfig struct {
+	CtrlClient ctrlclient.Client
+	Logger     micrologger.Logger
+}
+
+type Conditions struct {
+	ctrlClient ctrlclient.Client
+	logger     micrologger.Logger
+}
+
+var (
+	clusterStatus = prometheus.NewDesc(
+		prometheus.BuildFQName(MetricsNamespace, "cluster", "status"),
+		"Latest cluster status conditions as provided by the Cluster CR status.",
+		[]string{
+			"cluster_id",
+			"release_version",
```

Comment:
I took inspiration from [the metrics exposed in `cluster-operator`](https://github.com/giantswarm/cluster-operator/tree/master/service/collector). But I'm not sure why the `release_version` is necessary or wanted here? Should we keep it?

Reply:
might be used in case there are known bugs in certain releases in order to be able to silence alerts until customer upgrades maybe?

Reply:
These will be used for inhibitions, so I don't think that's a valid use case for these?